upload_multiple_files_to_single_record.py

```diff
@@ -0,0 +1,200 @@
+    list similar to the hocr example.
+    """
+    mime_type = mimetypes.guess_type(file_name)[0]
+    if not mime_type:
+        if file_name.endswith('.hocr'):
+            mime_type = 'text/vnd.hocr+html'
+        else:
+            return
+
+    return create_datafield('FFT__',
+                            [('a', presigned_response["data"]["fields"]['key']),
+                             ('c', local_checksum),
+                             ('e', mime_type),
+                             ('l', OBJECT_STORE_NAME),
+                             ('n', file_name)])
+
+
+def upload_metadata(string_xml):
+    """
+    Used the record API to upload metadata together with the FFT tag.
+    The FFT tag will link the file to the record.
+    """
+    url = SITE_URL + 'api/v1/record?&mode=insertorreplace&callback_email=' + CALLBACK_EMAIL
+    headers = {'Authorization': 'Token ' + API_KEY,
+               'Content-Type': 'application/xml'}
+
+    r = requests.post(url, headers=headers, data=string_xml)
+
+    print("Metadata upload status: %s" % (r.text,))
+
+
+def upload_and_save_xml(collection):
+    """
+    Upload the MARCXML with the record API and save a local copy of the file.
+    """
+    # Convert to string, replace
+    string_xml = tostring(collection, pretty_print=True, encoding="unicode")
+
+    # Replace capital TAG to lowercase tag in data fields
+    string_xml = string_xml.replace('datafield TAG="', 'datafield tag="')
+
+    string_xml_name = data_folder.parent / 'export_{}.xml'.format(data_folder.name)
+
+    # Save xml file
+    with open(string_xml_name, 'w') as xmlfile:
+        xmlfile.write(string_xml)
+
+    # Upload xml file
+    try:
+        upload_metadata(string_xml)
+    except UnicodeEncodeError as e:
+        print("Cannot import XML file: %s. Error message: \n %s" % (string_xml_name, e))
+
+
+if __name__ == '__main__':
+    # Get the path to the files.
+    folder_path = input('Insert folder path for the files to upload:\n')
+    folder_path = folder_path.replace('\\', '').strip()
+    data_folder = Path(folder_path)
+    if not data_folder.is_dir():
+        print('The provided path is not a folder. Try again!')
+        exit()
+
+    # Create xml record object
+    new_record = E.record()
+
+    # Append metadata to record
+    for key, val in APPEND_METADATA.items():
+        if len(key) == 6:
+            new_record.append(
+                create_datafield(key[0:5],
+                                 [(key[5], val)]))
+
+    # Loop through all files in path, except DS_Store (MacOS specific files).
+    for i, file_path in enumerate(sorted(data_folder.rglob('*.[!DS_Store]*'))):
+
+        # Ignore directories
+        if file_path.is_dir():
+            continue
+
+        # Step 1: Get AWS presigned object
+        try:
+            presigned_response = request_presigned_object()
+        except ConnectionResetError:
+            print("Presign - File failed to upload due: %s" % (file_path,))
```

Comment:
This isn't really a file failed to upload error: there might be a problem with getting the presigned url that has nothing to do with the file? Instead of going to the next file, it should try to get another presigned post and use the same file (i.e. don't continue to the next file).